src/libsystemd-network/dhcp-network.c

```diff
@@ -104,9 +104,9 @@ static int _bind_raw_socket(int ifindex, union sockaddr_union *link,
                 .sll_protocol = htobe16(ETH_P_IP),
                 .sll_ifindex = ifindex,
                 .sll_hatype = htobe16(arp_type),
-                .sll_halen = mac_addr_len,
+                .sll_halen = bcast_addr_len,
         };
-        memcpy(link->ll.sll_addr, bcast_addr, mac_addr_len);
+        memcpy(link->ll.sll_addr, bcast_addr, bcast_addr_len);
```

Comment:
hmm, binding to the broadcast addr? can you explain?

Reply:
It's a raw/packet socket, and the broadcast address in this case is the hardware broadcast address (6 bytes of 0xFF in case of ethernet, something rather complex and not set in stone in the case of IB).
The initial discover is broadcasted out via this socket. Once the DHCP server is known normal UDP unicast is used.